lib/tool_shed/util/admin_util.py

```diff
@@ -89,27 +89,8 @@ def create_role(self, trans, **kwd):
                 ok = False
             else:
                 # Create the role
-                role = trans.app.model.Role(name=name, description=description, type=trans.app.model.Role.types.ADMIN)
-                trans.sa_session.add(role)
-                # Create the UserRoleAssociations
-                for user in [trans.sa_session.query(trans.app.model.User).get(x) for x in in_users]:
-                    ura = trans.app.model.UserRoleAssociation(user, role)
-                    trans.sa_session.add(ura)
-                # Create the GroupRoleAssociations
-                for group in [trans.sa_session.query(trans.app.model.Group).get(x) for x in in_groups]:
-                    gra = trans.app.model.GroupRoleAssociation(group, role)
-                    trans.sa_session.add(gra)
-                if create_group_for_role_checked:
-                    # Create the group
-                    group = trans.app.model.Group(name=name)
-                    trans.sa_session.add(group)
-                    # Associate the group with the role
-                    gra = trans.model.GroupRoleAssociation(group, role)
-                    trans.sa_session.add(gra)
-                    num_in_groups = len(in_groups) + 1
-                else:
-                    num_in_groups = len(in_groups)
-                trans.sa_session.flush()
+                role, num_in_groups = trans.app.security_agent.create_role(
```

Comment:
Toolshed does not have this method in its security agent (which is different than Galaxy's). Should be fixed in https://github.com/galaxyproject/galaxy/pull/6026/commits/40d0d997d38ea6832c83a78c880db0af5c23a90a

@scholtalbers